perfkitbenchmarker/virtual_machine.py

```diff
@@ -285,36 +314,65 @@ def RemoteCopy(self, file_path, remote_path='', copy_to=True):
                     (retcode, full_cmd, stdout, stderr))
       raise errors.VmUtil.SshConnectionError(error_text)
 
+  def _PushLongRunningCommandScripts(self):
+    """Pushes the scripts required by LongRunningRemoteCommand to this VM.
+
+    If the scripts have already been placed on the VM, this is a noop.
+    """
+    with self._has_remote_command_script_lock:
+      if not self._has_remote_command_script:
+        for f in (EXECUTE_COMMAND, WAIT_FOR_COMMAND):
+          self.PushDataFile(f, os.path.join(vm_util.VM_TMP_DIR,
+                                            os.path.basename(f)))
+
   def LongRunningRemoteCommand(self, command):
     """Runs a long running command on the VM in a robust way.
 
+    Executes a long running command via a pair of scripts on the VM:
+
+    * EXECUTE_COMMAND, which runs 'command' in a nohupped background process.
+    * WAIT_FOR_COMMAND, which waits on a file lock held by EXECUTE_COMMAND until
+      'command' completes, then returns with the stdout, stderr, and exit status
+      of 'command'.
+
+    Temporary SSH failures (where ssh returns a 255) will be tolerated.
+
     Args:
       command: A valid bash command.
 
     Returns:
       A tuple of stdout and stderr from running the command.
     """
+    self._PushLongRunningCommandScripts()
+
+    execute_path = os.path.join(vm_util.VM_TMP_DIR,
+                                os.path.basename(EXECUTE_COMMAND))
+    wait_path = os.path.join(vm_util.VM_TMP_DIR,
+                             os.path.basename(WAIT_FOR_COMMAND))
+
     uid = uuid.uuid4()
-    stdout_file = '/tmp/stdout%s' % uid
-    stderr_file = '/tmp/stderr%s' % uid
-    long_running_cmd = ('nohup %s 1> %s 2> %s &' %
-                        (command, stdout_file, stderr_file))
-    self.RemoteCommand(long_running_cmd)
-    get_pid_cmd = 'pgrep %s' % command.split()[0]
-    pid, _ = self.RemoteCommand(get_pid_cmd)
-    pid = pid.strip()
-    check_process_cmd = ('if ! ps -p %s >/dev/null; then echo "Stopped"; fi' %
-                         pid)
-    while True:
-      stdout, _ = self.RemoteCommand(check_process_cmd)
-      if stdout.strip() == 'Stopped':
-        break
-      time.sleep(60)
-
-    stdout, _ = self.RemoteCommand('cat %s' % stdout_file)
-    stderr, _ = self.RemoteCommand('cat %s' % stderr_file)
+    file_base = os.path.join(vm_util.VM_TMP_DIR, 'cmd%s' % uid)
+    stdout_file = file_base + '.stdout'
+    stderr_file = file_base + '.stderr'
+    status_file = file_base + '.status'
 
-    return stdout, stderr
+    if not isinstance(command, basestring):
+      command = ' '.join(command)
+
+    start_command = ['nohup', 'python', execute_path,
```

Comment:
I noticed "python" in this line, if this is a generic OS command, why do we need to involve python here?